tests/components/fronius/test_init.py

```diff
@@ -53,3 +57,54 @@ async def test_inverter_error(
     ):
         config_entry = await setup_fronius_integration(hass)
         assert config_entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_inverter_night_rescan(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test dynamic adding of an inverter discovered automatically after a Home Assistant reboot during the night."""
+    mock_responses(aioclient_mock, fixture_set="igplus_v2", night=True)
+    config_entry = await setup_fronius_integration(hass, is_logger=True)
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    # Only expect logger during the night
+    fronius_entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(fronius_entries) == 1
+
+    # Switch to daytime
+    mock_responses(aioclient_mock, fixture_set="igplus_v2", night=False)
+    async_fire_time_changed(
+        hass, dt_util.utcnow() + timedelta(minutes=SOLAR_NET_RESCAN_TIMER)
+    )
+    await hass.async_block_till_done()
+
+    # We expect our inverter to be present now
+    device_registry = dr.async_get(hass)
+    inverter_1 = device_registry.async_get_device(identifiers={(DOMAIN, "203200")})
+    assert inverter_1.manufacturer == "Fronius"
+
+    # After another re-scan we still only expect this inverter
+    async_fire_time_changed(
+        hass, dt_util.utcnow() + timedelta(minutes=SOLAR_NET_RESCAN_TIMER * 2)
+    )
+    await hass.async_block_till_done()
+    inverter_1 = device_registry.async_get_device(identifiers={(DOMAIN, "203200")})
+    assert inverter_1.manufacturer == "Fronius"
+
+
+async def test_inverter_rescan_interruption(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test test interruption of re-scan during runtime to only cause a warning."""
+    mock_responses(aioclient_mock, fixture_set="igplus_v2", night=False)
+    config_entry = await setup_fronius_integration(hass, is_logger=True)
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    with patch(
+        "pyfronius.Fronius.inverter_info",
+        side_effect=FroniusError,
+    ):
+        async_fire_time_changed(
+            hass, dt_util.utcnow() + timedelta(minutes=SOLAR_NET_RESCAN_TIMER)
+        )
+        await hass.async_block_till_done()
```

Comment:
can we assert something after that time-change? Maybe the device count didn't change.

Reply:
Improved on that one a bit. Added setup, error and re-attempt assertions.